test/e2e/notification_test.go

```diff
@@ -0,0 +1,78 @@
+package e2e
+
+import (
+	"testing"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/wait"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestNotifications(t *testing.T) {
+	suite.Run(t, &notificationTestSuite{})
+}
+
+type notificationTestSuite struct {
+	suite.Suite
+	namespace   string
+	ctx         *framework.Context
+	awaitility  *wait.Awaitility
+	hostAwait   *wait.HostAwaitility
+	memberAwait *wait.MemberAwaitility
+}
+
+func (s *notificationTestSuite) SetupSuite() {
+	notificationList := &v1alpha1.NotificationList{}
+	s.ctx, s.awaitility = testsupport.WaitForDeployments(s.T(), notificationList)
+	s.hostAwait = s.awaitility.Host()
+	s.memberAwait = s.awaitility.Member()
+	s.namespace = s.awaitility.HostNs
+}
+
+func (s *notificationTestSuite) TearDownTest() {
+	s.ctx.Cleanup()
+}
+
+func (s *notificationTestSuite) TestNotificationCleanup() {
+
+	// Create and approve "janedoe"
+	janedoeName := "janedoe"
+	janeSignup := createAndApproveSignup(s.T(), s.awaitility, janedoeName)
+
+	s.T().Run("notification created and deleted", func(t *testing.T) {
+
+		hostAwait := wait.NewHostAwaitility(s.awaitility)
+		memberAwait := wait.NewMemberAwaitility(s.awaitility)
+
+		// Get the latest signup version
+		userSignup, err := s.awaitility.Host().WaitForUserSignup(janeSignup.Name)
+		require.NoError(t, err)
+
+		// First, wait for the MasterUserRecord to exist, no matter what status
+		mur, err := hostAwait.WaitForMasterUserRecord(userSignup.Status.CompliantUsername)
+		require.NoError(t, err)
+
+		// Then wait for the associated UserAccount to be provisioned
+		userAccount, err := memberAwait.WaitForUserAccount(mur.Name,
+			wait.UntilUserAccountHasConditions(provisioned()))
+
+		_, err = hostAwait.WaitForMasterUserRecord(mur.Name,
+			wait.UntilMasterUserRecordHasConditions(provisioned(), provisionedNotificationCRCreated()))
```

Comment:
We don't really need to wait for userSignup (it's done when we do `createAndApproveSignup()` and UserAccounts. Waiting for the mur to be provisioned is enough. UserID, username, etc are also available in the MUR.
```suggestion
		// Then wait for the MUR to be provisioned
		mur, err = hostAwait.WaitForMasterUserRecord(mur.Name,
			wait.UntilMasterUserRecordHasConditions(provisioned(), provisionedNotificationCRCreated()))
```

Reply:
@alexeykazakov makes sense - is this more what you were looking for: e81dddd9dc778bdd808eecba8e30335074db22cb?

Reply:
yep